test/users/engin/compiler/nestedRecordField.future

```diff
@@ -0,0 +1,4 @@
+bug: inner record object as a field
+
+when outer is a record it seems to fail whereas when it is class it seems to be
+working fine
```

Comment:
Your test case is different here, is that intentional?  In the other test you were accessing the outer type's field from a method on the inner type, but here you're accessing the inner type's field from a method on the outer type.  I don't think it'll make a difference in terms of allowing the code to pass, but it's something to be aware of.